relay-server/src/envelope/mod.rs

```diff
@@ -1221,7 +1231,7 @@ mod tests {
             );
             *Envelope::parse_bytes(bytes).unwrap()
         };
-        envelope.set_dsc(dsc.clone());
+        envelope.headers.set_dsc(dsc.clone());
```

Comment:
Test wouldn't have to change:

```suggestion
        envelope.set_dsc(dsc.clone());
```